packages/@o3r/store-sync/src/sync-storage/storage-sync.ts

```diff
@@ -0,0 +1,338 @@
+import { INIT, UPDATE } from '@ngrx/store';
+import { deepFill } from '@o3r/core';
+import type { Logger } from '@o3r/core';
+import type { StorageKeyConfiguration, StorageKeys, SyncStorageConfig, SyncStorageSyncOptions } from './interfaces';
+
+/**
+ * Reviver the date from a JSON field if the string is matching iso format. Return the same value otherwise
+ *
+ * @param _key JSON item key name
+ * @param value JSON item value
+ */
+export const dateReviver = (_key: string, value: any) => {
+  if (typeof value === 'string' && /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(value)) {
```

Comment:
What about dates with timezone ?

Reply:
I took it from the original code but it is actually a good point :S

Reply:
After checking, this is mainly due to the `toJSON()` serialization.
The date is serialized with a format matching this PR (Without local timezone), in case of serialization including timezone, a custom deserializer needs to be provided